packages/e2e.sandbox.subscribeapp/src/MapComponent.tsx

```diff
@@ -0,0 +1,177 @@
+    ObjectCoordinatesMap
+  >({});
+
+  const initializeEntity = (info: EntityInformation) => {
+    setEntityInfo((prev) => ({
+      ...prev,
+      [info.id]: info,
+    }));
+  };
+
+  const deleteEntity = (info: EntityInformation) => {
+    setEntityInfo((prev) => {
+      const newEntityInfo = { ...prev };
+      delete newEntityInfo[info.id];
+      return newEntityInfo;
+    });
+  };
+
+  const updateCoordinates = (primaryKey: string, coordinates: Coordinates) => {
+    if (!(primaryKey in entityInfo)) {
+      return;
+    }
+    setObjectCoordinates((prev) => ({
+      ...prev,
+      [primaryKey]: coordinates,
+    }));
+  };
+
+  const fetchInitialPoints = async () => {
+    const page = await foundryClient($Objects.MtaBus).fetchPageWithErrors({
+      $pageSize: 50,
+    });
+    if (isOk(page)) {
+      const objects = page.value.data;
+
+      for (const object of objects) {
+        const primaryKey = object.$primaryKey;
+        initializeEntity({
+          id: primaryKey,
+          routeId: object.routeId
+            ?? "",
+          nextStopId: object.nextStopId ?? "",
+        });
+      }
+      for (const object of objects) {
+        const primaryKey = object.$primaryKey;
+        object.positionId?.getLatestValue().then(
+          (latestPosition) => {
+            console.log("Latest position is", latestPosition);
+            if (latestPosition != null) {
+              const newCoordinates = {
+                lat: latestPosition?.value.coordinates[1]!,
+                lng: latestPosition?.value.coordinates[0]!,
+              };
+              updateCoordinates(primaryKey, newCoordinates);
+            }
+          },
+        );
+      }
+    }
+  };
+
+  React.useEffect(() => {
+    fetchInitialPoints();
+  }, []);
+
+  foundryClient($Objects.MtaBus)[__EXPERIMENTAL__NOT_SUPPORTED_YET_subscribe](
+    ["nextStopId", "positionId", "routeId", "vehicleId"],
+    {
+      onChange: (object) => {
+        const primaryKey = object.object.$primaryKey;
+
+        if (!(primaryKey in entityInfo)) {
+          return;
+        }
+
+        console.log(
+          "Received update for",
+          primaryKey,
+          ". Length of map is ",
+          Object.keys(entityInfo).length,
+        );
+
+        if (object.state === "REMOVED") {
+          deleteEntity(entityInfo[primaryKey]);
+        }
+
+        if (object.object.positionId != null) {
+          const newCoordinates = {
+            lat: object.object.positionId?.lastFetchedValue?.value
+              .coordinates[1]!,
+            lng: object.object.positionId?.lastFetchedValue?.value
+              .coordinates[0]!,
+          };
+
+          updateCoordinates(primaryKey, newCoordinates);
+        }
+
+        initializeEntity({
+          id: primaryKey,
+          routeId: object.object.routeId ?? entityInfo[primaryKey].routeId,
+          nextStopId: object.object.nextStopId
+            ?? entityInfo[primaryKey].nextStopId,
+        });
```

Comment:
I have multiple thoughts on this. This seems like the ideal way for a user to update their cache since the properties are coming back as `| undefined`, and they can just fill in the blanks with what is already.  Is it possible a user just does `!` on the returned object, then they receive a reference update, and then it all breaks? I would like to hope that developers follow simple practices such as good handling here, do we need to account for those who won't? However, I am liking the decision we made to include reference updates alongside the rest of these as if you do have proper error handling AND you set up your cache in the right way, the best practice becomes clear. However, the second part of the AND is the bigger worry to me